src/main/webapp/app/shared/metis/metis.service.ts

```diff
@@ -259,42 +266,21 @@ export class MetisService {
     }
 
     /**
-     * sorts posts by two criteria
-     * 1. criterion: displayPriority is PINNED -> pinned posts come first
-     * 2. criterion: displayPriority is ARCHIVED  -> archived posts come last
-     * 3. criterion: vote-emoji count -> posts with more vote-emoji counts comes first
-     * 4. criterion: creationDate -> most recent comes at the end (chronologically from top to bottom)
-     * @return Post[] sorted array of posts
+     * creates empty default post that is needed on initialization of a newly opened modal to edit or create a post
      */
-    static sortPosts(posts: Post[]): Post[] {
-        return posts.sort(function (postA, postB) {
-            const postAVoteEmojiCount = postA.reactions?.filter((reaction) => reaction.emojiId === VOTE_EMOJI_ID).length ?? 0;
-            const postBVoteEmojiCount = postB.reactions?.filter((reaction) => reaction.emojiId === VOTE_EMOJI_ID).length ?? 0;
-            if (postA.displayPriority === DisplayPriority.PINNED && postB.displayPriority !== DisplayPriority.PINNED) {
-                return -1;
-            }
-            if (postA.displayPriority !== DisplayPriority.PINNED && postB.displayPriority === DisplayPriority.PINNED) {
-                return 1;
-            }
-            if (postA.displayPriority === DisplayPriority.ARCHIVED && postB.displayPriority !== DisplayPriority.ARCHIVED) {
-                return 1;
-            }
-            if (postA.displayPriority !== DisplayPriority.ARCHIVED && postB.displayPriority === DisplayPriority.ARCHIVED) {
-                return -1;
-            }
-            if (postAVoteEmojiCount > postBVoteEmojiCount) {
-                return -1;
-            }
-            if (postAVoteEmojiCount < postBVoteEmojiCount) {
-                return 1;
-            }
-            if (Number(postA.creationDate) > Number(postB.creationDate)) {
-                return 1;
-            }
-            if (Number(postA.creationDate) < Number(postB.creationDate)) {
-                return -1;
-            }
-            return 0;
-        });
+    createEmptyPostForContext(courseWideContext?: CourseWideContext, exercise?: Exercise, lectureId?: number): Post {
+        const emptyPost: Post = new Post();
+        if (courseWideContext) {
+            emptyPost.courseWideContext = courseWideContext;
+            emptyPost.course = { id: this.courseId } as Course;
```

Comment:
You are right, I wanted to keep the objects small but, the naming indicates, that some could assess the properties of a course via `post.course`. I decided to assign the course object instead of only setting the id. During serialization and deserialization I only consider the id of the course anyways, so the payload should not be too high when the post is transported.

Reply:
I'd also be fine with only the id being stored, as long as it is stated clearly. Storing the whole course also works, of course